### PR TITLE
Frontend experiment

### PR DESCRIPTION
**Pull request by aaronfng**
Main changes:

- Reformatting/refactoring
- Adding example HTML files for testing our plugin
- Tracking the text nodes between frontend and backend using unique IDs. Not the most efficient implementation but works
- Minor unit-test and CI additions

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -3,3 +3,6 @@ venv
 __pycache__
 .idea
```
Comment (mhoare) [inline .gitignore]: needs to be `.idea*`

**Comment by mhoare** [comment]
I think should be alright

**Comment by mhoare** [comment]
Please also remove all the idea and other dotfiles

File: .gitignore (hunk 1 above)
Comment (aaronfng) [reply .gitignore]: .idea or idea/ ignores it, * isn't needed. All the .idea files in this branch are excluded.

**Comment by aaronfng** [comment]
Think the only dot file is .coverage, removed

**Comment by mhoare** [comment]
Looks Good

**Pull request merged**